agent.go

```diff
@@ -41,6 +41,15 @@ type agent struct {
 	driverCancelFuncs map[string][]func()
 }
 
+func (c *controller) GossipClusterMembers() []string {
+	c.Lock()
+	defer c.Unlock()
+	if c.agentInitDone == nil {
```

Comment:
Can you make this check for this `if c.agent != nil && c.agentInitDone == nil` instead? I know this code will work correctly today based on the current implementation but to make this check robust and decoupled for future code changes we should only try to get `Members` only if `c.agent` is not nil.